src/mds/MDLog.cc

```diff
@@ -1432,6 +1433,8 @@ void MDLog::_replay_thread()
     }
 
     logger->set(l_mdl_rdpos, pos);
+    logger->set(l_mdl_expos, journaler->get_expire_pos());
+    logger->set(l_mdl_wrpos, journaler->get_write_pos());
```

Comment:
Should we update these earlier ? What if the replay loop takes a long time ? Then in the perf dump we still couldn't see it ?

Reply:
Can these be updated +once_ before we enter the replay loop and then the relevant fields in the loop itself?

Reply:
> Should we update these earlier ? What if the replay loop takes a long time ? Then in the perf dump we still couldn't see it ?

I think that's an issue only if the log event's replay gets stuck for the **first** log event.

Reply:
> Can these be updated +once_ before we enter the replay loop and then the relevant fields in the loop itself?

I think I was just trying to keep these neat alongside the `rdpos` updates. The other concern is `standby-replay` daemon replay where the journaler write/expire pos is always changing. However, the standby-replay daemon should break out of this loop regularly anyway.

Reply:
Fair enough.